nan.h

```diff
@@ -1668,13 +1668,15 @@ class Callback {
 
 /* abstract */ class AsyncWorker {
  public:
-  explicit AsyncWorker(Callback *callback_)
+  explicit AsyncWorker(Callback *callback_,
+                       const char* resource_name = "nan:AsyncWorker")
       : callback(callback_), errmsg_(NULL) {
     request.data = this;
 
     HandleScope scope;
     v8::Local<v8::Object> obj = New<v8::Object>();
     persistentHandle.Reset(obj);
+    async_resource = new AsyncResource(obj, resource_name);
```

Comment:
Could this not  be a direct member of the class or am I missing something?

Reply:
`AsyncResource` doesn't have a default constructor. It needs the resource object which is not constructed until later in `AsyncWorker` constructor here.